app/connectors/ExternalPropertyLinkConnector.scala

```diff
@@ -0,0 +1,99 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import binders.GetPropertyLinksParameters
+import http.VoaHttpClientAuth
+import http.VoaHttpClient
+import javax.inject.{Inject, Named}
+import models.modernised.{ClientPropertyLink, CreatePropertyLink, OwnerPropertyLink, PropertyLinksWithAgents, PropertyLinksWithClient}
+import uk.gov.hmrc.http._
+import models.ModernisedEnrichedRequest
+import uk.gov.hmrc.play.config.ServicesConfig
+import models.PaginationParams
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class ExternalPropertyLinkConnector @Inject() //(@Named("VoaNewAuthedBackendHttp") http: VoaHttpClientAuth,
+                                              (@Named("VoaAuthedBackendHttp") val http: VoaHttpClient,
+                                              @Named("voa.myOrganisationsPropertyLinks") myOrganisationsPropertyLinksUrl: String,
+                                              @Named("voa.myOrganisationsPropertyLink") myOrganisationsPropertyLinkUrl: String,
+                                              @Named("voa.myClientsPropertyLinks") myClientsPropertyLinksUrl: String,
+                                              @Named("voa.myClientsPropertyLink") myClientsPropertyLinkUrl: String,
+                                              @Named("voa.createPropertyLink") createPropertyLinkUrl: String,
+                                               conf: ServicesConfig) (implicit executionContext: ExecutionContext)
+  extends JsonHttpReads with OptionHttpReads with RawReads {
+
+
+
+
+  def getMyOrganisationsPropertyLinks(searchParams: GetPropertyLinksParameters, params: Option[PaginationParams])
+                                     (implicit hc: HeaderCarrier, request: ModernisedEnrichedRequest[_]): Future[Option[PropertyLinksWithAgents]] = {
+
+    println(myOrganisationsPropertyLinksUrl)
+    http.GET[Option[PropertyLinksWithAgents]](
+      myOrganisationsPropertyLinksUrl,
+      modernisedPaginationParams(params) ++
+        List(searchParams.address.map("address" -> _),
+          searchParams.baref.map("baref" -> _),
+          searchParams.agent.map("agent" -> _),
+          searchParams.status.map("status" -> _),
+          searchParams.sortfield.map("sortfield" -> _),
+          searchParams.sortorder.map("sortorder" -> _)).flatten)
+  }
+
+
+  def getMyOrganisationsPropertyLink(submissionId: String)
+                                    (implicit hc: HeaderCarrier, request: ModernisedEnrichedRequest[_]): Future[Option[OwnerPropertyLink]] =
+    http.GET[Option[OwnerPropertyLink]](myOrganisationsPropertyLinkUrl.replace("{propertyLinkId}", submissionId))
```

Comment:
@PeterPerhac has moved a templating library into HMRC domain that we really should be used.

Reply:
Do you have the repo details of the library?